src/v/raft/tests/raft_fixture.h

```diff
@@ -0,0 +1,298 @@
+
+    ss::future<> stop();
+
+private:
+    template<typename ReqT, typename RespT>
+    ss::future<result<RespT>> dispatch(model::node_id, ReqT req);
+    ss::gate _gate;
+    absl::flat_hash_map<model::node_id, std::unique_ptr<channel>> _channels;
+    raft_node_map& _nodes;
+};
+
+/**
+ * Node hosting Raft replica
+ */
+class raft_node_instance : public ss::weakly_referencable<raft_node_instance> {
+public:
+    using leader_update_clb_t
+      = ss::noncopyable_function<void(leadership_status)>;
+    explicit raft_node_instance(
+      model::node_id id,
+      model::revision_id revision,
+      raft_node_map& node_map,
+      leader_update_clb_t leader_update_clb);
+
+    raft_node_instance(const raft_node_instance&) = delete;
+    raft_node_instance(raft_node_instance&&) noexcept = delete;
+    raft_node_instance& operator=(raft_node_instance&&) = delete;
+    raft_node_instance& operator=(const raft_node_instance&) = delete;
+    ~raft_node_instance() = default;
+
+    ss::lw_shared_ptr<consensus> raft() { return _raft; }
+
+    ss::future<> start(
+      std::vector<raft::vnode> initial_nodes,
+      std::optional<raft::state_machine_manager_builder> builder
+      = std::nullopt);
+    ss::future<> stop();
+
+    ss::future<> remove_data();
+
+    void leadership_notification_callback(leadership_status);
+
+    model::ntp ntp() {
+        return {
+          model::ns("test_ns"),
+          model::topic_partition(
+            model::topic(fmt::format("node_{}", _id)), model::partition_id(0))};
+    }
+
+    vnode get_vnode() const { return {_id, _revision}; }
+
+    ss::future<ss::circular_buffer<model::record_batch>>
+    read_all_data_batches();
+
+private:
+    model::node_id _id;
+    model::revision_id _revision;
+    ss::sstring _base_directory;
+    ss::shared_ptr<in_memory_test_protocol> _protocol;
+    ss::sharded<storage::api> _storage;
+    std::chrono::milliseconds _election_timeout = 500ms;
+    ss::sharded<features::feature_table> _features;
+    ss::sharded<coordinated_recovery_throttle> _recovery_throttle;
+    recovery_memory_quota _recovery_mem_quota;
+    std::unique_ptr<heartbeat_manager> _hb_manager;
+    leader_update_clb_t _leader_clb;
+    prefix_logger _logger;
+    ss::lw_shared_ptr<consensus> _raft;
+    bool started = false;
+};
+
+class raft_fixture
+  : public seastar_test
+  , public raft_node_map {
+public:
+    using raft_nodes_t = absl::
+      flat_hash_map<model::node_id, std::unique_ptr<raft_node_instance>>;
+    static constexpr raft::group_id group_id = raft::group_id(123);
+
+    std::optional<std::reference_wrapper<raft_node_instance>>
+    node_for(model::node_id id) final {
+        auto it = _nodes.find(id);
+        if (it == _nodes.end()) {
+            return std::nullopt;
+        }
+        return *it->second;
+    };
+
+    raft_node_instance& add_node(model::node_id id, model::revision_id rev);
+    ss::future<> stop_node(model::node_id id, bool remove_data_dir = false);
```

Comment:
nit for remove_data_dir can we use `ss::bool_class` or something? It helps keep the tests readable.